hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -243,6 +252,16 @@ public String getPrimaryGroup() {
 
   @Override
   public void close() throws IOException {
+    for (SelfRenewingLease lease : leaseRefs.keySet()) {
```

Comment:
This likely to take time? I'm worried about what happens if there's network problems and this gets invoked. Ideally this would be done in parallel, but abfs doesnt (yet) have a thread pool